### PR TITLE
add loop option

### PR DESCRIPTION
**Pull request by weslayer**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -165,7 +165,10 @@ async def play(url: str):
         url_type = _get_url_type(url)
         # Check if the given url is a valid video or playlist
         if url_type == UrlType.VIDEO:
-            threading.Thread(target=handle_play, args=(url,)).start()
+            if loop:
+                threading.Thread(target=handle_play, args=(url,loop)).start()
+            else:
+                threading.Thread(target=handle_play, args=(url,False)).start()
```
Comment (evanugarte) [inline server.py]: why have an if/else? we can just start the thread and pass in `loop`

**Review by evanugarte — COMMENTED** [review]
lets change the checkbox to be a button on the same row as all the other inputs like

image

and then when toggled, the button will change text (and color slightly) and look like
image

File: server.py (hunk 1 above)
Comment (weslayer) [reply server.py]: done

File: static/index.html [hunk 2]
```diff
@@ -33,14 +34,32 @@ <h3 id="title"></h3>
...
+        
         async function Play(e) {
             e.preventDefault();
             const url = document.getElementById('url').value;
             let playURL = new URL(window.location.pathname + "play", window.location.origin)
-
+            
             playURL.searchParams.append("url", encodeURIComponent(url))
+            if(loopBool){
+                playURL.searchParams.append("loop","True")
```
Comment (evanugarte) [inline static/index.html]: as long as something is in the param the api should loop, the below works

```suggestion
                playURL.searchParams.append("loop", 1)
```

File: static/index.html [hunk 3]
```diff
@@ -33,14 +34,32 @@ <h3 id="title"></h3>
         let stateURL = new URL(window.location.pathname + "state", window.location.origin)
         let stopURL = new URL(window.location.pathname + "stop", window.location.origin)
         let playButton = document.getElementById("play");
+        let loopButton = document.getElementById("loop");
+        let loopBool = false;
```
Comment (evanugarte) [inline static/index.html]: lets call it this instead
```suggestion
        let videoWillLoop = false;
```

File: static/index.html [hunk 4]
```diff
@@ -33,14 +34,32 @@ <h3 id="title"></h3>
...
         playButton.addEventListener("click", Play);
-   
+        loopButton.addEventListener("click", toggleBool);
+
+        async function toggleBool(){
```
Comment (evanugarte) [inline static/index.html]:
```suggestion
        loopButton.addEventListener("click", toggleLoopVideo);

        async function toggleLoopVideo(){
```

File: static/index.html [hunk 5]
```diff
@@ -33,14 +34,32 @@ <h3 id="title"></h3>
...
+            loopBool = !loopBool;
+            if(loopBool){
+                loopButton.textContent = "Video will loop!";
+                loopButton.style.backgroundColor="darkblue";
+            }
+            else{
```
Comment (evanugarte) [inline static/index.html]:
```suggestion
            } else {
```

**Review by evanugarte — COMMENTED** [review]
were almost there nice button

File: static/index.html (hunk 5 above)
Comment (weslayer) [reply static/index.html]: done

File: static/index.html (hunk 4 above)
Comment (weslayer) [reply static/index.html]: done

File: static/index.html (hunk 3 above)
Comment (weslayer) [reply static/index.html]: done

File: static/index.html (hunk 2 above)
Comment (weslayer) [reply static/index.html]: done

**Review by evanugarte — APPROVED** [review]

**Pull request merged**